sql/courses.sql

```diff
@@ -0,0 +1,67 @@
+DROP TABLE IF EXISTS planner_courses;
+DROP TABLE IF EXISTS scheduler_sections;
+DROP TABLE IF EXISTS sections;
+DROP TABLE IF EXISTS courses;
+DROP TABLE IF EXISTS instructors;
+DROP TABLE IF EXISTS grades_types;
+
+CREATE TABLE courses (
+    course_id       INT           NOT NULL AUTO_INCREMENT,
+    year            YEAR          NOT NULL, -- e.g. 2018
+    term            TINYINT       NOT NULL, -- FA: 1, WI: 2, SP: 3, SU: 4
+    department      VARCHAR(30)   NOT NULL, -- e.g. CS
+    course_number   VARCHAR(10)   NOT NULL, -- e.g. 124
+    name            VARCHAR(150)  NOT NULL, -- e.g. Operating Systems
+    description     TEXT, -- Currently unused
+    units_lecture   FLOAT         NOT NULL, -- e.g. 3
+    units_lab       FLOAT         NOT NULL, -- e.g. 6
+    units_homework  FLOAT         NOT NULL, -- e.g. 3
+    units           FLOAT         AS (units_lecture + units_lab + units_homework),
+    PRIMARY KEY (course_id),
+    UNIQUE (year, term, department, course_number)
+);
+
+CREATE TABLE instructors (
+    instructor_id  INT          NOT NULL AUTO_INCREMENT,
+    instructor     VARCHAR(60)  NOT NULL, -- e.g. Pinkston, D
+    PRIMARY KEY (instructor_id),
+    UNIQUE (instructor)
+);
+
+CREATE TABLE grades_types (
+    grades_type_id  INT          NOT NULL AUTO_INCREMENT,
+    grades_type     VARCHAR(30)  NOT NULL, -- e.g. PASS-FAIL
+    PRIMARY KEY (grades_type_id)
+);
+
+CREATE TABLE sections (
+    course_id       INT           NOT NULL,
+    section_number  TINYINT       NOT NULL, -- e.g. 1
+    instructor_id   INT           NOT NULL,
+    grades_type_id  INT           NOT NULL,
+    times           VARCHAR(100)  NOT NULL, -- e.g. MWF 13:00 - 13:55
+    locations       VARCHAR(100)  NOT NULL, -- e.g. 213 ANB; currently unused
```

Comment:
If `locations` is unused, why not allow `NULL`?

Reply:
The upload script always puts the data in there, it just isn't used by the scheduler or planner. I figured we might as well store this data since it is included in the registrar's file.